### PR TITLE
Add integration for SSD (read & write targets)

### PR DESCRIPTION
**Pull request by trevorwhitney**
**What this PR does / why we need it**:

Adds an integration test for Simple Scalable Deployment (SSD), which comprises just read and write targets.

### CODE REVIEW

**Review by dannykopping — APPROVED** [review]
:heart: nice!

How comprehensive are we expecting to go with these? The PR description doesn't hint at what we're trying to achieve here. I love that we're starting with this, though.

For example:
This will currently only test in-memory logs in the head block; can you think of a way that we could ensure that the data is persistable to disk?
And should we test other critical features of SSD like ring communication / replication, etc?

**Comment by trevorwhitney** [comment]
@dannykopping great questions. I think we want to be conservative in expanding these tests because of how expensive they are. At this level, this test makes sure that if we make a change to a component, and forget to enable it or propagate that change to SSD mode, we will have an early warning sign. I would hold off expanding the scope of the test until we run into a bug that is related to persisting data to disk, ring communication / replication, etc. before expanding the test.

File: integration/loki_micro_services_delete_test.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func TestMicroServicesDeleteRequest(t *testing.T) {
 
 	require.NoError(t, clu.Run())
 
-	tenantID := randStringRunes(12)
+	tenantID := randStringRunes()
```
Comment (trevorwhitney) [inline integration/loki_micro_services_delete_test.go:68]: The linter was complaining that this is always called with 12

**Comment by dannykopping** [comment]
> @dannykopping great questions. I think we want to be conservative in expanding these tests because of how expensive they are. At this level, this test makes sure that if we make a change to a component, and forget to enable it or propagate that change to SSD mode, we will have an early warning sign. I would hold off expanding the scope of the test until we run into a bug that is related to persisting data to disk, ring communication / replication, etc. before expanding the test.

Sounds good to me.
What aspect is expensive right now? Are the tests quite slow?

**Comment by trevorwhitney** [comment]
> What aspect is expensive right now? Are the tests quite slow?

They are slower than the unit tests, but I think the real risk is the potential for bloat here. Every additional integration test adds more time than the addition of a unit test.

**Pull request merged**